homeassistant/components/sensor/__init__.py

```diff
@@ -508,12 +508,14 @@ def state(self) -> Any:  # noqa: C901
             _LOGGER.warning(
                 "Entity %s (%s) is using state class '%s' which "
                 "is impossible considering device class ('%s') it is using; "
+                "expected one of %s; "
                 "Please update your configuration if your entity is manually "
                 "configured, otherwise %s",
                 self.entity_id,
                 type(self),
                 state_class,
                 device_class,
+                [str(value) if value else "no state class" for value in classes],
```

Comment:
This is a bit weird, as `DEVICE_CLASS_STATE_CLASSES` never has` None` value.
Should it not instead be:
```suggestion
                [value for value in classes] if classes else ["no state class"],
```

Reply:
Or maybe:
```suggestion
                [str(value) if value else "no state class" for value in classes or {None}],
```

Reply:
Actually, maybe it is fine:
- it will show "[]" if it's an empty set
- it is future-proof in case `None` even gets added to `DEVICE_CLASS_STATE_CLASSES` (though I can't remember why it is not typed  `dict[SensorDeviceClass, set[SensorStateClass]]`)

Reply:
Sorry - me again.

Based on this discussion https://github.com/home-assistant/core/pull/84402#discussion_r1071169863 I think the type hint should be adjusted to be `DEVICE_CLASS_STATE_CLASSES: dict[SensorDeviceClass, set[SensorStateClass]] = {`

And then you can use:
```suggestion
                [value.value for value in classes],
```

Reply:
Made it more beautiful! Thanks for the input 👍